design/20210203.certificate-request-identity.md

```diff
@@ -0,0 +1,165 @@
+---
+title: Certificate Request Identity
+authors:
+  - "@joshvanl"
+reviewers:
+  - "@joshvanl"
+approvers:
+  - "@joshvanl"
+editor: "@joshvanl"
+creation-date: 2021-02-03
+last-updated: 2021-02-03
+status: provisional
+---
+
+# Identity
+
+## Table of Contents
+
+TODO
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [API Changes](#api-changes)
+  - [Upgrading](#upgrading)
+  - [Test Plan](#test-plan)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Certificate Identity Pass Down](#certificate-identity-pass-down)
+<!-- /toc -->
+
+## Summary
+
+In order to implement policy and improve auditing introspection of certificates
+requested and signed by cert-manager, cert-manager must implement a better
+identity framework. Identity in this context is a user or machine which requests
+a signed certificate from cert-manager. This identity will be managed and
+enforced on CertificateRequest resources by cert-manager.
+
+
+## Motivation
+
+The identity of a requester forms a large part of an automated policy system
+that needs to decide whether a certificate should or shouldn't be signed. By
+creating a trusted source of identity for `CertificateRequest`s, a policy engine
+is able to determine whether that identity is allowed to request that
+certificate, given some policy configuration setup by a cluster administrator.
+
+Although auditing exists in Kubernetes and exposes the identity of the
+requester, its configuration is not always exposed to end users, such as
+when using managed Kubernetes (GKE, EKS etc). In scenarios where
+configuring auditing is available, it is often not preferable and an
+anti-pattern to make runtime decisions on historical audit logs.
+
+The upstream [Kubernetes certificates
+`CertificateSigningRequest`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#certificatesigningrequest-v1-certificates-k8s-io)
+API has identity as part of the resource. As the project intends to align and
+likely make this resource compatible with cert-manager in some way, the
+`CertificateRequest` should expose identical identity details. This means
+extensions or additions to cert-manager (policy, auditing) can have the same
+guarantees about what information is available from both resources.
+
+
+### Goals
+
+- Introduce identity to the `CertificateRequest` resource which is securely
```

Comment:
At what point should we start referring to this as 'UserInfo' rather than identity? I'm cautious that in a sense, it's not technically an identity, rather its information about the user? Not too sure if the distinction matters too much but just my thoughts as I read through the doc top to bottom :)

Reply:
Yeah, not too bothered. Think the word got stuck in my head somehow and kept repeating it :joy: